examples/fluids/qfunctions/newtonian.h

```diff
@@ -803,25 +736,164 @@ CEED_QFUNCTION(IFunction_Newtonian)(void *ctx, CeedInt Q,
+  const CeedScalar cv = context->cv;
+  const CeedScalar Rd = cp - cv;
+  const CeedScalar gamma = cp / cv;
+
+  CeedPragmaSIMD
+  // Quadrature Point Loop
+  for (CeedInt i=0; i<Q; i++) {
+    // -- Interp-to-Interp q_data
+    const CeedScalar wdetJ      =   q_data[0][i];
+    // -- Interp-to-Grad q_data
+    // ---- Inverse of change of coordinate matrix: X_i,j
+    // *INDENT-OFF*
+    const CeedScalar dXdx[3][3] = {{q_data[1][i],
+                                    q_data[2][i],
+                                    q_data[3][i]},
+                                   {q_data[4][i],
+                                    q_data[5][i],
+                                    q_data[6][i]},
+                                   {q_data[7][i],
+                                    q_data[8][i],
+                                    q_data[9][i]}
+                                  };
+    // *INDENT-ON*
+
+    CeedScalar U[5], kmstress[6], Tau_d[3] __attribute((unused));
+    for (int j=0; j<5; j++) U[j] = jac_data[j][i];
+    for (int j=0; j<6; j++) kmstress[j] = jac_data[5+j][i];
+    for (int j=0; j<3; j++) Tau_d[j] = jac_data[5+6+j][i];
+    const CeedScalar x_i[3] = {x[0][i], x[1][i], x[2][i]};
+    State s = StateFromU(context, U, x_i);
+
+    CeedScalar dU[5], dx0[3] = {0};
+    for (int j=0; j<5; j++) dU[j] = dq[j][i];
+    State ds = StateFromU_fwd(context, s, dU, x_i, dx0);
+
+    State grad_ds[3];
+    for (int j=0; j<3; j++) {
+      CeedScalar dUj[5];
+      for (int k=0; k<5; k++) dUj[k] = Grad_dq[0][k][i] * dXdx[0][j]
+                                         + Grad_dq[1][k][i] * dXdx[1][j]
+                                         + Grad_dq[2][k][i] * dXdx[2][j];
+      grad_ds[j] = StateFromU_fwd(context, s, dUj, x_i, dx0);
+    }
+
+    CeedScalar dstrain_rate[6], dkmstress[6], stress[3][3], dstress[3][3], dFe[3];
+    KMStrainRate(grad_ds, dstrain_rate);
+    NewtonianStress(context, dstrain_rate, dkmstress);
+    KMUnpack(dkmstress, dstress);
+    KMUnpack(kmstress, stress);
+    ViscousEnergyFlux_fwd(context, s.Y, ds.Y, grad_ds, stress, dstress, dFe);
+
+    StateConservative dF_inviscid[3];
+    FluxInviscid_fwd(context, s, ds, dF_inviscid);
+
+    // Total flux
+    CeedScalar dFlux[5][3];
+    for (int j=0; j<3; j++) {
+      dFlux[0][j] = dF_inviscid[j].density;
+      for (int k=0; k<3; k++)
+        dFlux[k+1][j] = dF_inviscid[j].momentum[k] - dstress[k][j];
+      dFlux[4][j] = dF_inviscid[j].E_total + dFe[j];
+    }
+
+    for (int j=0; j<3; j++) {
+      for (int k=0; k<5; k++) {
+        Grad_v[j][k][i] = -wdetJ * (dXdx[j][0] * dFlux[k][0] +
+                                    dXdx[j][1] * dFlux[k][1] +
+                                    dXdx[j][2] * dFlux[k][2]);
+      }
+    }
+
+    const CeedScalar dbody_force[5] = {0, ds.U.density *g[0], ds.U.density *g[1], ds.U.density *g[2], 0};
+    for (int j=0; j<5; j++)
+      v[j][i] = wdetJ * (context->ijacobian_time_shift * dU[j] - dbody_force[j]);
+
+    if (1) {
```

Comment:
Is this hardcoded `if` necessary? Maybe it should be this:
```suggestion
    if (context->stabilization != STAB_NONE) {
```

Reply:
It's inactive for Galerkin because the computed tau is zero. I kept the `if` block for development convenience and because I think that block of code can be refactored to be a bit cleaner and more efficient. I'd prefer that control flow continue to pass through this even for `-stab none` unless we want to optimize for Galerkin (in which case we'd want to create a separate qfunction that passes less data).